src/main/java/com/walmartlabs/x12/common/AbstractStandardX12Parser.java

```diff
@@ -0,0 +1,118 @@
+/**
+Copyright (c) 2018-present, Walmart, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+package com.walmartlabs.x12.common;
+
+import com.walmartlabs.x12.X12Parser;
+import com.walmartlabs.x12.X12Segment;
+import com.walmartlabs.x12.exceptions.X12ParserException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+
+public abstract class AbstractStandardX12Parser<T extends AbstractStandardX12Document> implements X12Parser<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractStandardX12Parser.class);
+
+    public static final String ISA_ID = "ISA";
+    public static final String GROUP_SEGMENT_ID = "GS";
+
+    /**
+     * template for parsing a standard EDI X12 document
+     *
+     * @return T (the Class associated with the parser)
+     * @throws X12ParserException
+     */
+    @Override
+    public T parse(String sourceData) {
+        AbstractStandardX12Document x12Doc = null;
+
+        if (!StringUtils.isEmpty(sourceData)) {
+            x12Doc = this.createX12Document();
```

Comment:
Should we wrap this method with `try..catch` block and throw `X12ParserException` from `catch`?